### DISCUSSION TITLE
ERROR WITH COUNT FUNCTION IN QFIELD 2.7.6

### DISCUSSION THREAD

**Discussion by CACG1022** [opengisch/QField#4252] [Bugs]
I am currently doing a project which automatically counts and enumerates each marker that is added to a polygon. the project works fine in Qgis but in Qfield it doesn't work fine anymore. in 2.7.1 versions of Qfield it works fine but in 2.7.6 it no longer works. The expression that I am using is the following which is inserted in the field called CONSECUTIVE. attach the project with the videos:

### NOTE: in Qgis 3.30 it works fine, but in Qfield 2.7.6 it doesn't work

**PROJECT:**
[MAPP8.zip](https://github.com/opengisch/QField/files/11373031/MAPP8.zip)

**Test in QFIELD 2.7.6**

https://user-images.githubusercontent.com/118937011/235665975-ef11f691-b5c6-4a3a-b940-d72fb73f7f2e.mp4

**Test in QGIS 3.30**

https://user-images.githubusercontent.com/118937011/235666059-702107cc-6018-455c-b563-328f2197e4ef.mp4

**This is the expression used:**

![Captura](https://user-images.githubusercontent.com/118937011/235666415-f46bd185-7905-4521-8da9-cf0ca1e7327e.JPG)

**Comment by nirvn** [comment]
@CACG1022 , use this expression as default value for your CONSECUTIVE field:

```
with_variable('poli',aggregate(layer:='POLIGONOS',aggregate:='concatenate',
expression:=to_string("NOMBRE"),
filter:=intersects( $geometry, geometry(@parent))),
aggregate(layer:='AGREGAR',aggregate:='count',
expression:="CONSECUTIV",filter:="POLIGONO"=@poli)+1
)
```

With that, you don't need to activate [x] apply default value on update, which allows you to edit your features without changing that initial value. It also works on QField >= 2.7.6 :)
